### PR TITLE
CICD: Add GHA trigger for PR workflow of master branch

### LINKED ISSUE #2699: Empty pull requests
@tlimoncelli may I ask what the reason is for these empty (closed) pull requests?

1. https://github.com/StackExchange/dnscontrol/pull/2698
2. https://github.com/StackExchange/dnscontrol/pull/2679
3. https://github.com/StackExchange/dnscontrol/pull/2675
4. https://github.com/StackExchange/dnscontrol/pull/2634
5. https://github.com/StackExchange/dnscontrol/pull/2556
6. https://github.com/StackExchange/dnscontrol/pull/2352

### PR DESCRIPTION
**Pull request by cafferata**
Trigger the GitHub Actions pull request workflow by branch name `tlim_testpr`. This fixes #2699 and saves you having to create empty (closed) pull requests.

See also:
1. The [Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore) documentation.
2. The GitHub Actions workflow running action [#7414124200](https://github.com/cafferata/dnscontrol/actions/runs/7414124200) by the branch `tlim_testpr` on my GitHub fork repository.

The GitHub Actions `PR: Run all tests` workflow can be found here:
1. [StackExchange/dnscontrol/actions/workflows/pr_test.yml?query=branch:tlim_testpr](https://github.com/StackExchange/dnscontrol/actions/workflows/pr_test.yml?query=branch%3Atlim_testpr).
    * Fork [cafferata/dnscontrol](https://github.com/cafferata/dnscontrol/actions/workflows/pr_test.yml?query=branch%3Atlim_testpr) example.
2. [StackExchange/dnscontrol/actions/workflows/pr_test.yml?query=event:push](https://github.com/StackExchange/dnscontrol/actions/workflows/pr_test.yml?query=event%3Apush).
    * Fork [cafferata/dnscontrol](https://github.com/cafferata/dnscontrol/actions/workflows/pr_test.yml?query=event%3Apush) example.

What do you think? Could this help you further?

### CODE REVIEW

**Comment by tlimoncelli** [comment]
I'm intrigued!

If I understand this right.... any time I want to try a test PR, I just push to that branch?  I assume something like `git checkout master && git push origin test_pr` ?

**Comment by cafferata** [comment]
> If I understand this right.... any time I want to try a test PR, I just push to that branch?

That's correct!

> I assume something like git checkout master && git push origin test_pr ?

Yes, also correct! 😄

**Comment by cafferata** [comment]
Or even shorter with an update to date master.

```shell
git push origin master:tlim_testpr --force
```

File: .github/workflows/pr_test.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 name: "PR: Run all tests"
 on:
+  push:
+    branches:
+      - 'tlim_testpr'
```
Comment (tlimoncelli) [inline .github/workflows/pr_test.yml:5]: plz add a comment that `git push origin master:tlim_testpr --force` will trigger a full PR test on the master branch.

**Comment by cafferata** [comment]
Done, including a deep link to the GitHub Actions ☺️

**Pull request merged**